src/cxy_cache.erl

```diff
@@ -363,8 +363,10 @@ fetch_item(Cache_Name, Key) ->
 %% Fetch from Generation 1 or 2, after updating with a newer key version or leaving existing newest key version.
 refresh_item(Cache_Name, Key) ->
     Found_Fn     = fun(Fn_Cache_Name, _Value, _Version, Fn_New_Gen_Id, Fn_Mod, Fn_Key) ->
-                           Fn_New_Cached_Value = create_new_value(Fn_Cache_Name, Fn_New_Gen_Id, Fn_Mod,Fn_Key),
-                           return_cache_refresh(Fn_Cache_Name, Fn_New_Cached_Value)
+                           case create_new_value(Fn_Cache_Name, Fn_New_Gen_Id, Fn_Mod,Fn_Key) of
+                               no_value_available                  -> return_cache_refresh(Fn_Cache_Name, no_value_available);
+                               {_New_Version, Fn_New_Cached_Value} -> return_cache_refresh(Fn_Cache_Name, Fn_New_Cached_Value)
```

Comment:
I wish these were wrapped in a record - that way you could give the static analysis tools a chance to catch stuff like this